x-pack/plugins/maps/server/content_management/maps_storage.ts

```diff
@@ -0,0 +1,316 @@
+  constructor() {}
+
+  async get(ctx: StorageContext, id: string): Promise<MapGetOut> {
+    const {
+      utils: { getTransforms },
+      version: { request: requestVersion },
+    } = ctx;
+    const transforms = getTransforms(cmServicesDefinition, requestVersion);
+    const soClient = await savedObjectClientFromRequest(ctx);
+
+    // Save data in DB
+    const {
+      saved_object: savedObject,
+      alias_purpose: aliasPurpose,
+      alias_target_id: aliasTargetId,
+      outcome,
+    } = await soClient.resolve<MapSavedObjectAttributes>(SO_TYPE, id);
+
+    const response: MapGetOut = {
+      item: savedObjectToMapItem(savedObject, false),
+      meta: {
+        aliasPurpose,
+        aliasTargetId,
+        outcome,
+      },
+    };
+
+    // Validate DB response and DOWN transform to the request version
+    const { value, error: resultError } = transforms.get.out.result.down<MapGetOut, MapGetOut>(
+      response
+    );
+
+    if (resultError) {
+      throw Boom.badRequest(`Invalid response. ${resultError.message}`);
+    }
+
+    return value;
+  }
+
+  async bulkGet(): Promise<never> {
+    // Not implemented. Maps does not use bulkGet
+    throw new Error(`[bulkGet] has not been implemented. See MapsStorage class.`);
+  }
+
+  async create(
+    ctx: StorageContext,
+    data: MapCreateIn['data'],
+    options: CreateOptions
+  ): Promise<MapCreateOut> {
+    const {
+      utils: { getTransforms },
+      version: { request: requestVersion },
+    } = ctx;
+    const transforms = getTransforms(cmServicesDefinition, requestVersion);
+
+    // Validate input (data & options) & UP transform them to the latest version
+    const { value: dataToLatest, error: dataError } = transforms.create.in.data.up<
+      MapSavedObjectAttributes,
+      MapSavedObjectAttributes
+    >(data);
+    if (dataError) {
+      throw Boom.badRequest(`Invalid data. ${dataError.message}`);
+    }
+
+    const { value: optionsToLatest, error: optionsError } = transforms.create.in.options.up<
+      CreateOptions,
+      CreateOptions
+    >(options);
+    if (optionsError) {
+      throw Boom.badRequest(`Invalid options. ${optionsError.message}`);
+    }
+
+    // Save data in DB
+    const soClient = await savedObjectClientFromRequest(ctx);
+    const savedObject = await soClient.create<MapSavedObjectAttributes>(
+      SO_TYPE,
+      dataToLatest,
+      optionsToLatest
+    );
+
+    // Validate DB response and DOWN transform to the request version
+    const { value, error: resultError } = transforms.create.out.result.down<
+      MapCreateOut,
+      MapCreateOut
+    >({
+      item: savedObjectToMapItem(savedObject, false),
+    });
+
+    if (resultError) {
+      throw Boom.badRequest(`Invalid response. ${resultError.message}`);
+    }
+
+    return value;
+  }
+
+  async update(
+    ctx: StorageContext,
+    id: string,
+    data: MapUpdateIn['data'],
+    options: UpdateOptions
+  ): Promise<MapUpdateOut> {
+    const {
```

Comment:
This code is repeated a few times. Could `ctx` provide a `getTransforms` fn that only takes the `cmServicesDefinition`?

Reply:
It could be improved indeed 👍 
I created https://github.com/elastic/kibana/issues/154648 to track this